### PR TITLE
Resilience v1.0

### PR DESCRIPTION
**Pull request by dephillipsmichael**
This PR is mostly exposing ORK classes publicly and some more functionality with already exposed classes.  

I wrote some magical classes in Resilience that allow completely custom UI for ORKFormStepViewController and ORKQuestionStepViewController while still using 100% of their functionality.  I accomplished this by hooking up corresponding cell and UI element events and feeding those into the existing ORKFSVC or ORQSVC methods.

### CODE REVIEW

File: ResearchKit/Common/ORKFormStepViewController.m [hunk 1]
```diff
@@ -613,6 +613,18 @@ - (BOOL)allAnsweredFormItemsAreValid {
     return YES;
 }
 
+- (BOOL)isFormItemValidFor:(NSString*)formItemIdentifier {
```
Comment (dephillipsmichael) [inline ResearchKit/Common/ORKFormStepViewController.m:616]: This is the only added functionality.  I created it specifically so I could check if a field's error message was no longer applicable.

File: ResearchKit/Common/ORKQuestionStepViewController.h [hunk 2]
```diff
@@ -52,7 +53,42 @@ NS_ASSUME_NONNULL_BEGIN
...
+
+/**
+ * @return the current answer stored, can be nil
+ */
+- (id<NSCopying, NSCoding, NSObject>)answer;
+
+/**
+ * @param the answer to store
+ */
+- (void)setAnswer:(id<NSCopying,NSObject,NSCoding>)answer;
```
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKQuestionStepViewController.h]: Nit: could you please fix the spacing.

File: ResearchKit/Common/ORKSurveyAnswerCellForText.h [hunk 3]
```diff
@@ -31,17 +31,23 @@
...
+#import "ORKAnswerTextView.h"
+#import <ResearchKit/ORKTypes.h>
 
 NS_ASSUME_NONNULL_BEGIN
+ORK_CLASS_AVAILABLE
+@interface ORKSurveyAnswerCellForText : ORKSurveyAnswerCell <UITextViewDelegate>
 
-@interface ORKSurveyAnswerCellForText : ORKSurveyAnswerCell
+/**
+ * Allow exxternal classes to set textview's text for answer cell
```
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKSurveyAnswerCellForText.h]: typo exxternal = external

File: ResearchKit/Common/ORKSurveyAnswerCellForText.m [hunk 4]
```diff
@@ -41,17 +40,20 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
...
 @property (nonatomic, strong) ORKAnswerTextView *textView;
-
 @end
 
 
 @implementation ORKSurveyAnswerCellForText {
     NSInteger _maxLength;
 }
 
+- (void) setText:(NSString*)text {
```
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKSurveyAnswerCellForText.m]: Please change the spacing. Thanks.

**Review by syoung-smallwisdom — CHANGES_REQUESTED** [review]
Couple of nits. 

Could you please verify that running the units tests in `ORKTest` also pass. (Travis isn't currently set up to run those.)

Also, please check with Josh that this doesn't code stomp him. Thanks!

**Comment by dephillipsmichael** [comment]
Tests succeeded

**Pull request merged**